### ISSUE TITLE
Support More Granular Updating to Current API View

### ISSUE THREAD

**Issue by alzimmermsft** [Azure/azure-sdk-tools#3286]
While reviewing the Java Core API views this month I noticed that many of the API views haven't been updated for a few weeks. From discussions with @praveenkuttappan I was informed that a new API view will only be generated if a meaningful difference has been made. At least for Java, changing dependency versions isn't a meaningful difference (one that'll require approval before releasing) and for many libraries that will be the only difference that may occur for months. Unfortunately, this means useful information may be lost and could result in confusing API views to exist. For example [azure-core-serializer-json-jackson](https://apiview.dev/Assemblies/Review/83768c0e56094634b973c8de003c09de/7702690237e743cdb7134c4534280611?diffRevisionId=b9f6537c29444b88900bb3b1ef76c64c&doc=False&diffOnly=True) looks like it has had a recent API difference but looking at API view metadata and knowing the library it has been API stable for a few months, only receiving dependency updates.

My request is that API view should support more granular updating, instead of checking for a meaningful difference to update the latest API view it should be any difference in what is generated. So, for Java this would mean that upgrading dependencies would cause a new API view, to give reviewers the most up-to-date information as possible.

**Comment by JonathanGiles** [comment]
I don't think this is Java specific, at least not immediately. Assigning to @praveenkuttappan to get his thoughts.

**Comment by praveenkuttappan** [comment]
This will be a change in parser side to exclude sections from skip diff range tokens if dependency needs to be considered in revision comparison.

**Comment by JonathanGiles** [comment]
@alzimmermsft If I understand correctly, you would like the APIView tooling for Java to mark dependency version changes as breaking changes that should trigger a new generation?

**Comment by alzimmermsft** [comment]
I wasn't so much thinking version increments should be a breaking change but noteworthy enough to create a new ApiView

**Comment by tjprescott** [comment]
I think the issue here is that the CI itself looks at whether there are API changes when deciding to create a new revision. In this case, even though there's a diff in the codefile (on account of the difference in dependencies) the script that's making this decision doesn't _see_ that change.

How common is this in Java?
